docs/developer-docs/security/security-best-practices/dos.mdx

```diff
@@ -4,32 +4,72 @@ keywords: [security, concept, denial of service, dos, cycles, captcha]
+A Denial of Service (DoS) attack aims to make a system unavailable by overwhelming it with requests or data. A Distributed Denial of Service (DDoS) attack is a more sophisticated version, where the attack originates from multiple sources, making it harder to block. 
 
-* Expensive calls that only need to be called from other canisters can require some amount of cycles to be sent along with the call to compensate for the cycles consumed by the execution.
+In the case of canisters, such attacks can quickly deplete cycles, making the canister unable to process legitimate requests. Learn more about DoS in the [Security Best Practices: Denial of Service](https://internetcomputer.org/docs/current/developer-docs/security/security-best-practices/dos) guide.
 
-* Finally, it is also an option to charge for ingress messages, but that is not currently supported by the protocol itself and a custom solution would need to be designed.
+### Preventing DoS and DDoS Attacks
 
-## For expensive calls, consider using captchas or proof of work
+To protect your canisters from DoS and DDoS attacks, consider the following strategies:
+- **Bot prevention techniques**: Use methods like captchas or proof of work to ensure only legitimate users can access your canister. Captchas help verify that the user is human, while proof of work requires the user to spend computational resources to proceed, deterring automated attacks. An example implementation can be found in the [Internet Identity project](https://github.com/dfinity/internet-identity).
+- **Monitor cycles usage**: Regularly track your canisters cycles consumption and set alerts for any sudden spikes that may indicate an attack.
+- **Ingress message charging**: While charging for ingress messages (external requests to the canister) is not natively supported, custom solutions could be implemented to make sure that any expensive actions have costs associated with them.
 
-### Security concern
+## 2. Managing Resource Allocation
 
-If an update or query call is expensive in terms of memory used or cycles consumed, this may make it easy for bots to render the canister unusable (e.g. by filling up storage).
+### The "Noisy Neighbor" Problem
```

Comment:
Another strategy to deal with this problem is to have an efficient migration strategy to another subnet or at least have a multi canister architecture where you take into account the load on each subnet to decide where to deploy new canisters. Of course also taking into account reducing inter subnet calls as much as possible for canisters that talk a lot to each other.

Some dapps specifically avoid some of the known busy subnets or just target a subnet where current resource consumption is very low. Of course, you never know when some noisy dapp will deploy onto your subnet.

Reply:
agreed

Reply:
added